### PR TITLE
unserialized models can be incremented

### PR DESCRIPTION
**Pull request by tondef001**
### Summary

Clicking the plus button for unserialized models now properly increments the model and increases instock and total values.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. go to view all models
2. Press the plus for an unserialized model.
3. Modal text and toast text should reflect that the model is unserialized (shouldn't mention adding a new item)
4. Run tests and such

### Links

- [TFF-223](https://msoese.atlassian.net/browse/TFF-223)
- TheFourFifths/consus-client/pull/115

### CODE REVIEW

File: docs/REST.md [hunk 1]
```diff
@@ -202,6 +203,37 @@ Content-Type: application/json
...
 }
 ```
+
+## PATCH `/api/model/instock`
+
+Updates an unserialized model's total and instock values after adding a new one
+
+### Parameters
+
+* `modelAddress`: Address of the model to increment
```
Comment (mattmahn) [inline docs/REST.md]: Nit: we already know it's a model, why not just `address`?

File: docs/REST.md [hunk 2]
```diff
@@ -202,6 +203,37 @@ Content-Type: application/json
...
+
+### Sample Response
+
+```json
+{
+    "status": "success",
+    "data": {
+        "model": {
```
Comment (mattmahn) [inline docs/REST.md]: NACK: there is no `model`, just
```json
{
    "status": "success",
    "data": {
        "name": "Resistor",
        "and": "Et cetera"
    }
}
```

File: test/unit/store/model-store.js [hunk 3]
```diff
@@ -261,4 +261,22 @@ describe('ModelStore', () => {
             assert.strictEqual(25, modifiedModel.inStock);
         });
     });
+
+    it('should increment an unserialized models in stock amount', () => {
+        return addAction('INCREMENT_STOCK', {
+            modelAddress: unserializedModel.address
+        }).then(() => {
+            let modifiedModel = ModelStore.getRecentlyUpdatedModel();
+            assert.strictEqual(21, modifiedModel.count);
```
Comment (mattmahn) [inline test/unit/store/model-store.js]: NACK: this and the next are backwards

File: test/functional/increment-unserialized-model.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                price: 10.50,
+                count: 20
+            }).then(actionId => {
+                model = ModelStore.getModelByActionId(actionId);
+            });
+        });
+    });
+
+    it('should increment a model', () => {
+        let newModel = {
```
Comment (mattmahn) [inline test/functional/increment-unserialized-model.js]: Nit: It think it'd be better if it was just `let newModel = model; newModel.count = 21; newModel.instock = 21;`

File: test/functional/increment-unserialized-model.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            assert.deepEqual(model, newModel);
+        });
+    });
+
+    it('should not accept a non-string address', () => {
+        return patch('model/instock', {
+        }).then(() => {
+            throw new Error('Unexpected success');
+        }).catch(e => {
+            assert.match(e.message, /A model address is required./);
```
Comment (mattmahn) [inline test/functional/increment-unserialized-model.js]: regex :+1:

File: docs/REST.md (hunk 1 above)
Comment (thofmann) [inline docs/REST.md]: We've been doing this in several places. I'd rather just use `address`, but it's not :consus: unless we're a little inconsistent.

File: docs/Actions.md [hunk 6]
```diff
@@ -110,6 +111,29 @@ Updates a model to have new attributes.
...
+    "name": "Resistor",
+    "description": "V = IR",
+    "manufacturer": "Pancakes R Us",
+    "vendor": "Mouser",
+    "location": "Shelf 14",
+    "price": 10.50,
+    "allowCheckout": "true",
+    "count": 21,
+    "inStock": 21,
+    "photo": "iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
```
Comment (thofmann) [inline docs/Actions.md]: image

File: src/store/model-store.js [hunk 7]
```diff
@@ -152,6 +152,17 @@ store.registerHandler('NEW_MODEL', data => {
     models.push(model);
 });
 
+store.registerHandler("INCREMENT_STOCK", data => {
+    let modelToInc = store.getModelByAddress(data.modelAddress);
```
Comment (thofmann) [inline src/store/model-store.js]: Nit: We still shouldn't be modifying store state via public getters.

File: docs/Actions.md (hunk 6 above)
Comment (mattmahn) [inline docs/Actions.md]: It's not even a JPEG either

File: docs/REST.md (hunk 1 above)
Comment (tondef001) [inline docs/REST.md]: The only time I use address is inside an actual model or item object.  Everywhere else I specify, obvious or not.

File: test/functional/increment-unserialized-model.js (hunk 5 above)
Comment (tondef001) [inline test/functional/increment-unserialized-model.js]: totally not copypasta

**Pull request merged**